src/blockCG.jl

```diff
@@ -106,10 +106,12 @@ for iter=1:maxIter
 	BLAS.gemm!('T','N',1.0,Q,Z,0.0,QTZ)
     Beta  = -pinvPTQ*QTZ
     
-    if ortho     
+    if ortho
         P,     =  mgs!(Z + P*Beta)
     else
-	    P     = Z  + P*Beta;
+		BLAS.gemm!('N','N',1.0,P,Beta,1.0,Z);
```

Comment:
You cannot change Z, because it might be R and then you're messing the
residual. See the correction that I made.

On Tue, Jun 7, 2016 at 11:55 AM, Lars Ruthotto <notifications@github.com>
wrote:

> In src/blockCG.jl
> <https://github.com/lruthotto/KrylovMethods.jl/pull/10#discussion_r66131364>
> :
>
> >          P,     =  mgs!(Z + P*Beta)
> >      else
> > -	    P     = Z  + P*Beta;
> > +		BLAS.gemm!('N','N',1.0,P,Beta,1.0,Z);
>
> well, if we left
>
> P[:] = Z
>
> in the else part of the if statement it should be the same, right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lruthotto/KrylovMethods.jl/pull/10/files/5fca412d06b6b13e45641cf84113379b10761c6c#r66131364>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AO5HPRHyelQ5jaPiadEKpkb1PWjByce2ks5qJb6TgaJpZM4IwM5i>
> .
>